ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/Listener.java

```diff
@@ -175,19 +175,28 @@ public synchronized boolean addConnection(Socket conn)
             // call to this method. it will have closed all the other connections
             // and will be waiting on this (listener) thread. so close this connection
             // before returning false
+            if (tsLogger.logger.isDebugEnabled()) {
+                tsLogger.logger.debug("Transaction listener is stopped. No new connection will be processed.");
+            }
             try {
                 conn.close();
             } catch (Exception e) {
-                // ignore
+                tsLogger.i18NLogger.warn_could_not_close_transaction_listener_connection(conn, e);
             }
             return false;
         }
     }
 
     public synchronized void removeConnection(Socket conn)
     {
-        connections.remove(conn);
-        notifyAll();
+        try {
+            connections.remove(conn);
```

Comment:
I think it would be clearer if the connections.remove was outside of this block as I don't think it will be throwing IOException, unless you think the notifyAll always done even with a problem on the linkedlist.remove() but if that is the case perhaps the conn.close() should be in it's own try block?

Reply:
agree. I will move the `connections.remove` outside of the block as it better corresponds with the prior functionality.